tests/test_viewController.py

```diff
@@ -0,0 +1,217 @@
+    vc.update_server_input(0, 180, False)
+    vc.main()
+    vc.update_server_input(25, 25, True)
+    vc.main()
+    return(vc.aim_coordinate[0], vc.aim_coordinate[1])
+
+def plot(p_long, p_lat, roll, yaw, pitch, theta, phi, lock_on, height):
+    """
+    This function plots the drone and its field of view. Red dot is
+    north, orange triangle is direction of drone, the X is where the
+    camera is aiming and blue triangle is the coordinate it focuses on,
+    if lock_on is true.
+    """
+    d_long = 15
+    d_lat = 15
+    if lock_on:
+        vc.update_server_input(theta, phi, False)
+        vc.main()
+    vc.update_fixhawk_input(roll, yaw, pitch, height, d_long, d_lat)
+    vc.update_server_input(theta, phi, lock_on)
+    vc.main()
+
+    #Target coordinate as a point
+    if lock_on:
+        coord_diff = (vc.aim_coordinate[0] - d_long,
+        vc.aim_coordinate[1] - d_lat)
+    else:
+        coord_diff = (p_long - d_long, p_lat - d_lat)
+
+    x_diff = np.tan(np.deg2rad(coord_diff[1])) * vc.earth_radius_at_lat(d_lat)
+    y_diff = np.tan(np.deg2rad(coord_diff[0])) * vc.earth_radius_at_lat(d_lat)
+
+    p_3dcoordinate = np.matrix([[x_diff], [y_diff], [-height]])
+
+    rot_matrix = vc.rotation_matrix(roll, yaw, pitch)
+    rot_matrix_inv = rot_matrix.transpose()
+
+    # Figur
+    fig = plt.figure()
+    ax = fig.gca(projection='3d', xlim=(-1 * SIZE, SIZE), ylim=(-1 * SIZE, SIZE),
+    zlim=(-SIZE,SIZE), autoscale_on = False, aspect = 'auto')
+    ax.set_ylabel('Drönarens riktning (y)')
+    ax.set_xlabel('Drönarens sidor (x)')
+
+    # Plotta kamerans sfär
+    u, v = np.mgrid[0:2*np.pi:20j, -np.pi/2:np.pi/2:20j]
+    x = (VIEW_SIZE)*np.cos(u)*np.sin(v)
+    y = (VIEW_SIZE)*np.sin(u)*np.sin(v)
+    z = -(VIEW_SIZE)*np.cos(v)
+    ax.plot_wireframe(x, y, z, color="r", alpha = 0.2)
+
+    # Plotta marken
+    point = np.array([[0], [0], [-height]])
+    point = rot_matrix_inv.dot(point)
+    normal = np.array([[0], [0], [1]])
+    normal = rot_matrix_inv.dot(normal)
+    p = np.array([point.item(0), point.item(1), point.item(2)])
+    n = np.array([normal.item(0), normal.item(1), normal.item(2)])
+    d = -p.dot(n)
+    xx, yy = np.meshgrid(range(-SIZE,SIZE+100,100), range(-SIZE,SIZE+100,100),
+    sparse = True)
+    z = (-n[0] * xx - n[1] * yy - d) * 1. /n[2]
+    ax.plot_surface(xx, yy, z, alpha = 0.5)
+
+    # Plotta sökt koordinat
+    p_3dcoordinate = rot_matrix_inv.dot(p_3dcoordinate)
+    ax.scatter(p_3dcoordinate.item(0), p_3dcoordinate.item(1),
+    p_3dcoordinate.item(2), marker = '^')
+
+    # plotta drönarens riktning
+    drone_dir = vc.angular_to_spherical(90, 0)
+
+    ax.scatter(VIEW_SIZE*drone_dir.item(0), VIEW_SIZE*drone_dir.item(1),
+    drone_dir.item(2)*VIEW_SIZE, marker = '^')
+    ax.scatter(0, 0, 0, marker = 'o')
+
+    # Norr
+    north = rot_matrix_inv.dot(drone_dir)
+    ax.scatter(VIEW_SIZE * north.item(0), VIEW_SIZE * north.item(1),
+    VIEW_SIZE * north.item(2), marker='o')
+
+    # Kamerans sikte
```

Comment:
No swedish comments please. Keep to one languge (multiple places)

Reply:
Fixed, see new commit "Resolved comments"